### PR TITLE
chore: add canary package generator

### PR DESCRIPTION
**Pull request by juanpicado**
**Type:** actions

**Description:**

- Publish a canary version of each PR
- Remove Node 10 and add Node 13 for testing
- Update GH Action main flow, lint before publish

### CODE REVIEW

**Comment by juanpicado** [comment]
take a look this @verdaccio/core-team 🚀 

It creates canary versions of each PR, also publish to specific registry and provides a comment with al CLI command just copy and install.

This is an experiment I'm running here in order to move it to Verdaccio, thus is much easier to test.

File: package.json [hunk 1]
```diff
@@ -171,7 +171,6 @@
     "lint:lockfile": "lockfile-lint --path yarn.lock --type yarn --validate-https --allowed-hosts verdaccio npm yarn",
     "coverage:publish": "codecov",
     "pre:webpack": "rimraf static/*",
-    "prepublish": "in-publish && npm run build || not-in-publish",
```
Comment (juanpicado) [inline package.json]: We don't need this. `prepublish` is deprecated anyway.

File: .github/workflows/canary.yml [hunk 2]
```diff
@@ -44,7 +44,7 @@ jobs:
         with:
           node-version: '12.x'
           registry-url: 'https://registry.verdaccio.org'
-      - run: npm publish
+      - run: npm publish -tag canary
```
Comment (zkochan) [inline .github/workflows/canary.yml]: should be --tag

Comment (juanpicado) [reply .github/workflows/canary.yml]: Thanks ! good catch !

File: .github/workflows/canary.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
+name: CI Canary
+
+on: [pull_request]
+
+jobs:
+  build_test_lint:
+    name: Node Smoke Test Befor Canary
```
Comment (priscilawebdev) [inline .github/workflows/canary.yml]:
```suggestion
    name: Node Smoke Test Before Canary
```

**Review by priscilawebdev — APPROVED** [review]

**Comment by priscilawebdev** [comment]
Very good @juanpicado let's move forward 🚀👏

**Pull request merged**